sklearnex/cluster/k_means.py

```diff
@@ -213,8 +204,8 @@ def _onedal_predict_supported(self, method_name, X, sample_weight=None):
                         "Supported data formats: Dense, CSR (oneDAL version >= 2024.7.0).",
                     ),
                     (
-                        _acceptable_sample_weights,
-                        "oneDAL doesn't support sample_weight, None or ones are acceptable",
+                        sample_weight is None,
+                        "oneDAL doesn't support sample_weight.",
```

Comment:
As the same question on the above. Did something changed on onedal side to update this message? Was this this a bug?

Reply:
Nothing changed in oneDAL, the checks were removed because they are causing significant perf regression. The message is more appropriate.